iis-common/src/test/java/eu/dnetlib/iis/common/java/jsonworkflownodes/SelectiveTestingConsumer.java

```diff
@@ -0,0 +1,144 @@
+ * Properties keys are paths to the fields we want to test in given object, properties values are expected values (with special markup for null: $NULL$).
+ * 
+ * Number of input avro records should be equal to the number of provided properties files.
+ *   
+ * @author mhorst
+ */
+public class SelectiveTestingConsumer implements Process {
+
+    private static final String PROPERTIES_CSV = "expectation_properties_csv";
+
+    private static final String PORT_INPUT = "datastore";
+
+    private static final String NULL_VALUE_INDICATOR = "$NULL$";
+
+    private final static Logger log = Logger.getLogger(SelectiveTestingConsumer.class);
+
+    private static final Map<String, PortType> inputPorts = new HashMap<String, PortType>();
+    
+    {
+        inputPorts.put(PORT_INPUT, new AnyPortType());
+    }
+    
+    //------------------------ LOGIC ---------------------------------
+    
+    @Override
+    public Map<String, PortType> getInputPorts() {
+        return inputPorts;
+    }
+
+    @Override
+    public Map<String, PortType> getOutputPorts() {
+        return Collections.emptyMap();
+    }
+
+    @Override
+    public void run(PortBindings portBindings, Configuration configuration, Map<String, String> parameters)
+            throws Exception {
+
+        Path inputRecordsPath = portBindings.getInput().get(PORT_INPUT);
+        String propertiesPathsCSV = parameters.get(PROPERTIES_CSV);
+        Preconditions.checkArgument(StringUtils.isNotBlank(propertiesPathsCSV), 
+                "no '%s' property value provided, field requirements were not specified!", PROPERTIES_CSV);
+
+        String[] recordsExpectationPropertiesLocations = StringUtils.split(propertiesPathsCSV, ',');
+        FileSystem fs = FileSystem.get(configuration);
+
+        if (!fs.exists(inputRecordsPath)) {
+            throw new RuntimeException(inputRecordsPath + " hdfs location does not exist!");
+        }
+        
+        try (CloseableIterator<SpecificRecord> recordsIterator = DataStore.getReader(new FileSystemPath(fs, inputRecordsPath))) {
+            
+            int recordsCount = 0;
+            
+            while (recordsIterator.hasNext()) {
+                
+                SpecificRecord record = recordsIterator.next();
+                recordsCount++;
+
+                if (recordsCount > recordsExpectationPropertiesLocations.length) {
+                    throw new RuntimeException("got more records than expected: " + "unable to verify record no " + recordsCount
+                            + ", no field specification provided! Record contents: " + JsonUtils.toPrettyJSON(record.toString()));
+                } else {
+                    validateRecord(record, readProperties(recordsExpectationPropertiesLocations[recordsCount - 1]));    
+                }
+            }
+            
+            if (recordsCount < recordsExpectationPropertiesLocations.length) {
```

Comment:
Marek, as far as I can see, you very often put 's' in the middle of the variable name. But, to be honest, it's not grammatically correct, so I would suggest you to not overuse it. For example, there is no such phrase in english like 'records count'. It should be 'record count' or 'count of records'. The same is true with record(s)Expectations etc.